### ISSUE TITLE
reactor.walker.i/o is not closed on stop

### ISSUE THREAD

**Issue by Mr-Bossman** [twisted/twisted#11602]
The following should be run after `reactor.stop()` or on exit but these pipes are left open.

```python3
os.close(reactor.waker.o)
os.close(reactor.waker.i)
```

The test code is as follows:

```python3
from twisted.internet import reactor
from twisted.internet.protocol import ClientFactory
import os

class fac(ClientFactory):
	def clientConnectionFailed(self, connector, reason):
		reactor.stop()

factory = fac()
reactor.connectTCP("localhost", 8000, factory)
reactor.run(False)
cmd = "ls -l /proc/"+str(os.getpid())+"/fd | grep pipe"
os.system(cmd)
os.close(reactor.waker.o)
os.close(reactor.waker.i)
print("And nothing after close:")
os.system(cmd)
```
And the output is:
```
lr-x------ 1 nobody nogroup 64 Jan 1 1970 4 -> pipe:[1]
l-wx------ 1 nobody nogroup 64 Jan 1 1970 5 -> pipe:[1]
And nothing after close:```

**Comment by adiroiban** [comment]
Thanks for the excellent bug report. Not sure what is going on there, but I was able to reproduce it on Ubuntu 22.04 with the default reactor.

**Comment by exarkun** [comment]
This looks like a duplicate of https://github.com/twisted/twisted/issues/3063 to me

**Comment by adiroiban** [comment]
yes. I think this can be closed as duplicate of #360 
and the work should be put into fixing the origin ticket.

Thanks for the extra info
